CMakeLists.txt

```diff
@@ -25,6 +25,7 @@ set( ENERGYPLUS_VERSION "${CMAKE_VERSION_MAJOR}.${CMAKE_VERSION_MINOR}.${CMAKE_V
 
 set( CMAKE_VERSION_BUILD "Unknown" CACHE STRING "Build number" )
 find_package(Git)
+find_package(OpenGL)
```

Comment:
@nealkruis Is this requiring that OpenGL be available to build? If so, wouldn't it be better to have a fallback that doesn't require the availability of OpenGL?

Reply:
I put this in here with the idea that I would disable some of the unit tests if the machine doesn't have OpenGL. I'm not sure it's really necessary though.

Reply:
Just to make sure we're all clear of the requirements on this.  

We will not make OpenGL, or any other graphical rendering library, a hard requirement for building or running EnergyPlus, on any platform.  We will need CMake logic in place to have fallbacks for building EnergyPlus where it doesn't attempt to find a rendering library.  And in EnergyPlus itself, we will need to have a switch for enabling specialized GPU-dependent calculations.  I would love if a user who doesn't have OpenGL could still take advantage of pixel counting, if not for performance, then for proper shading evaluation of complicated zone geometry.  With this in mind, I would like the input field for enabling this to specifically reflect "turning this on requires a graphics library on this machine", and then if E+ tries to run and doesn't find it, it gracefully fatals out and gives the user enough info to continue (either disable that feature or install the lib and re-run).

For the CI machines, as discussed yesterday, I am going to continue to push back against installing any additional libraries until we understand the consequences and packaging issues.  We have in the past installed stuff on the machines without thinking about about these consequences, and ended up having a difficult time getting packages back to normal ([Visual Studio drama of very late March 2017](https://github.com/NREL/EnergyPlus/commit/66a4a528058fc53760de51c7442fdf6ef26d529c)).  Once we have all these issues answered, we can decide how to proceed.  Eventually it seems important that we have a build on a clean machine that doesn't have any optional build dependencies on it, and then one that has all the optional build dependencies on it, but that would be a significant increase in burden, so I'm not sure we can completely support that.  We'll figure it out.

Just know that I am in support of this new feature, I am just going to be playing slow and tight with the dependencies and the builds.

Reply:
This makes sense to me. 

**Building:** It will likely require some preprocessor directives to disable portions of code if the build machine doesn't have the linking graphics libraries. We'll likely also want to put something in the `--version` info to indicate whether a build does not support graphics processing. I can also have CMake issue a nice loud warning if OpenGL isn't found.

**Running:** This is already handled, but the messaging could use clean up. We'll need different messages depending on why it fails:
1. the local machine doesn't have the required libraries, or
2. the build wasn't linked to a graphics library.

Reply:
Thanks @nealkruis, I should also point out that I don't expect any user to _ever_ encounter a build of EnergyPlus that they find to _not_ be built with support for the feature.  The builds we produce as packages _will_ have the capability built in so that we don't have stray packages lying around and confusion about whether a package has it or not.  

The build issue is purely for developers.  I don't want developers to have to install additional dependencies just to get a build going.  This is the same thing as how our build allows `BUILD_DOCS` and `BUILD_FORTRAN` so that if a developer doesn't need to build those features, they don't have to install the stuff to do it.  I imagine this is something that will be like `BUILD_FOR_OPENGL` or whatever, defaults to OFF, but can be flipped on easily.  And our CI build scripts will set `-DBUILD_FOR_OPENGL=ON`.  There's probably a niche case where an actual user wants to make a custom build that doesn't want this dependency also, but that would be the same as the developer use-case really.

No, I don't think BUILD_FOR_OPENGL is the right nomenclature :laughing: 

For user alert, how about:

```
$ ./energyplus --config
{
  "program_name": "EnergyPlus",
  "version_number": "9.2.0",
  "build_sha": "abc123",
  "architecture": "64bit",
  "compiled_with": "Visual Studio 2019",
  "compile_date": "UTC 8601 Date",
  "build_options": [
    "open_gl_dependency": "true/false",
  ]
}
```

How cool would that be!!??

Reply:
I might lean toward making builds with OpenGL by default (since most developer machines will have the necessary libraries). It's really only Linux machines where there will be a problem.

What if we have `BUILD_OPENGL_LINKS` (working nomenclature) `ON` by default. If it's ON and CMake can't find OpenGL, CMake issues an error with a suggestion that the user either install a working implementation or turn `BUILD_OPENGL_LINKS` to `OFF`.

Reply:
Yeah, I could be convinced to default it to ON.  In that case, I don't want it to be an advanced cmake variable so that it is easily accessible for disabling if someone should need to do it.  

`BUILD_OPENGL_LINKS` seems like an improvement.  We can keep noodling on that, but it seems to represent what we're doing.